File: week-6/go/membersys/repository/message_repository.go
package repository

import (
	. "Project/membersys/core"
)

type MessageRepository interface {
	AddMessage(memberId int, content string)
	GetMessages() []Message
}

func AddMessage(r MessageRepository, memberId int, content string) {
	r.AddMessage(memberId, content)
}

func GetMessages(r MessageRepository) []Message {
	return r.GetMessages()
}

File: week-7/py/member_system/repository/memory/message_repository.py
from member_system.core import List
from member_system.core import Message
from member_system.core import MessageRepository
from member_system.repository.memory.member_repository import MemoryMemberRepository

class MemoryMessageRepository(MessageRepository):
    def __init__(self, memberRepository: MemoryMemberRepository):
        self.memberRepository = memberRepository
        self.__db: List[Message] = []

    def addMessage(self, __memberId: int, __content: str) -> None:
        member = self.memberRepository.getMemberById(__memberId)
        self.__db.append(Message(member.name, __content))
    
    def getMessages(self) -> List[Message]:
        return self.__db
